client/network/src/service.rs

```diff
@@ -888,6 +888,43 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		});
 	}
 
+	/// High-level network status information.
+	pub async fn status(&self) -> Result<NetworkStatus<B>, RequestFailure> {
```

Comment:
```suggestion
	///
	/// Returns an error if the `NetworkWorker` is no longer running.
	pub async fn status(&self) -> Result<NetworkStatus<B>, ()> {
```

`RequestFailure` is specifically for sending a request to a peer.
I think using `()` for the error is fine.